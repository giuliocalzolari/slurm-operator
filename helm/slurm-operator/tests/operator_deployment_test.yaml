# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
---
suite: test operator deployment
templates:
- operator/deployment.yaml
chart:
  version: 1.2.3
  appVersion: 1.2.3
release:
  name: test-release
  namespace: test-namespace

tests:
- it: manifest should match snapshot
  asserts:
  - matchSnapshot: {}


- it: should add affinity
  set:
    operator:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: arch
                operator: In
                values:
                - x86_64
  asserts:
  - equal:
      path: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].values[0]
      value: x86_64


- it: should add tolerations
  set:
    operator:
      tolerations:
      - key: "test"
        operator: "Exists"
        effect: "NoExecute"
        tolerationSeconds: 6000
  asserts:
  - equal:
      path: spec.template.spec.tolerations[0].key
      value: test


- it: should set imagePullSecrets
  set:
    imagePullSecrets:
    - name: "test-docker-secret"
  asserts:
  - equal:
      path: spec.template.spec.imagePullSecrets[0].name
      value: test-docker-secret

- it: should set priorityClassName
  set:
    priorityClassName: "test-priority-class"
  asserts:
  - equal:
      path: spec.template.spec.priorityClassName
      value: test-priority-class


- it: should set replica count
  set:
    operator:
      replicas: 42
  asserts:
  - equal:
      path: spec.replicas
      value: 42

- it: should set namespace
  set:
    namespaceOverride: "custom-namespace"
  asserts:
  - equal:
      path: metadata.namespace
      value: custom-namespace
