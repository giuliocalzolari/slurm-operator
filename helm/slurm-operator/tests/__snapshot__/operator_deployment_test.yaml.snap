manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: slurm-operator
      namespace: test-namespace
    spec:
      replicas: 1
      revisionHistoryLimit: 0
      selector:
        matchLabels:
          app.kubernetes.io/instance: test-release
          app.kubernetes.io/name: slurm-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: test-release
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: slurm-operator
            app.kubernetes.io/version: 1.2.3
            helm.sh/chart: slurm-operator-1.2.3
        spec:
          containers:
            - args:
                - --accounting-workers
                - "4"
                - --controller-workers
                - "4"
                - --loginset-workers
                - "4"
                - --nodeset-workers
                - "4"
                - --restapi-workers
                - "4"
                - --token-workers
                - "4"
                - --slurmclient-workers
                - "2"
                - --zap-log-level
                - info
              image: ghcr.io/slinkyproject/slurm-operator:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
              name: slurm-operator
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
          hostname: slurm-operator
          priorityClassName: null
          serviceAccountName: test-release-slurm-operator
